source/main/terrain/TerrainObjectManager.cpp

```diff
@@ -1176,6 +1176,83 @@ void TerrainObjectManager::loadObject(const Ogre::String &name, const Ogre::Vect
 				continue;
 			}
 
+			if (!strncmp("spotlight", ptline, 9))
+			{
+				Vector3 lpos, ldir;
+				float lrange = 10, innerAngle = 45, outerAngle = 45;
+				ColourValue lcol;
+				String lname = "spotlight_" + TOSTRING(Math::RangeRandom(1000, 9999));

```

Comment:
Does `lname` have to be unique? If so then there is now guarantee here that it's unique.